projects/schematics/src/migrations/4_0/constructor-deprecations/data/navigation-ui.component.migration.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  SPARTACUS_STOREFRONTLIB,
+  HAMBURGER_MENU_SERVICE,
+  NAVIGATION_UI_COMPONENT,
+  NAVIGATION_UI_CONFIG,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const NAVIGATION_UI_COMPONENT_MIGRATION: ConstructorDeprecation = {
+  //projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts
+  class: NAVIGATION_UI_COMPONENT,
+  importPath: SPARTACUS_STOREFRONTLIB,
+  deprecatedParams: [],
```

Comment:
To make this migration work the deprecatedParams must describe shape of the previous contructor (router, renderer2, elementRef)